data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java

```diff
@@ -177,6 +179,16 @@ public static IndexConfiguration readIndexConfig(final PluginSetting pluginSetti
         if (templateFile != null) {
             builder = builder.withTemplateFile(templateFile);
         }
+
+        final String templateContent = pluginSetting.getStringOrDefault(TEMPLATE_CONTENT, null);
+        if (templateContent != null) {
+            builder = builder.withTemplateContent(templateContent);
+        }
+
+        if (templateContent != null && templateFile != null) {
+            LOG.warn("Both template_content and template_file are configured. Only template_content will be used");
```

Comment:
Was there a particular reason you have the content take precedence over the file?

Reply:
Not necessarily but if anything i am figuring the template_content will be more widely used